modules/common-jpa-impl/src/main/java/org/opencastproject/job/jpa/JpaJob.java

```diff
@@ -153,6 +155,9 @@
   @Column(name = "status")
   private int status;
 
+  @Column(name = "mediapackage")
```

Comment:
This requires a data migration, doesn't it?

Reply:
The job that represents the workflow is exactly the job that should have this mediapackage value set (it could be null for jobs that are operations as part of a workflow). The data migration will be a db conversion script using ExtractValue to get the mediapackage identified from the xml.

Reply:
Hi @staubesv 

Yes, as it stands this patch requires a data migration*.

I'm hoping to:
1. minimise the time spent performing an OC migration
2. make it "easier" to migrate events' true state... or at least make the migration process less destructive than what it currently happens to be (rebuilding indexes).

On your second point:
The direct link you refer to depends on the Solr index.

This patch attempts to remove Opencast's reliance on the Solr index for (almost) all things workflow related. The implication thereof means that the link between mediapackage -> workflow info via solr -> jobs breaks, hence the need for an alternative link between media packages and their associated jobs.

Also, only those jobs of operation START_WORKFLOW have a reference to their media package id (since actual workflows **are** jobs, but _not all_ jobs are workflows :)

Coming back to data migration*: using elasticsearch's current implementation, I could (in a bit of a roundabout way) get this patch to sidestep the need for data migration. But then I'm back to violating hope #2 above.
Either way, I plan on implementing some elasticsearch querying to minimise hitting the DB for every workflow request.

Reply:
Hi @slampunk, I have a *strong* request for the term "mediapackage" to be renamed to "mediapackageId" or "mpId" or something that implies that that value has no relationship to the mediapackage manifest and is only the id of that package.

Reply:
aren't job argument getting deleted at the end of workflows, IIRC there was a PR to to remove them as part of workflow cleanup? If so a separate mh_mediapackage_workflow (string  mpId; long jobId) table.

Reply:
From a database performance point of view, the fastest is probably a separate table, with 2nd-best adding the column to mh_job. I don't think adding it into mh_job_argument, as requires that require joins to get what they need with very large tables can be suboptimal (as seen with various asset manager db performance issues at present).

Reply:
Without digging into it: I would not use the oc_job_argument as the media package id is not really a job argument, is it? 
Without knowing the details, the most preferable solutions seems 1) to factor out workflows from the job table into an own table or 2) use a separate table to establish the relationship of media packages to workflows (mh_job_media_package consisting of the columns media_package_id and job_id) as @smarquard suggests.

Reply:
Yep, that should work well.